### DISCUSSION TITLE
R-ESRGAN-AnimeVideo-UI - Upscale videos with an easy to use interface

### DISCUSSION THREAD

**Discussion by Nenotriple** [xinntao/Real-ESRGAN#702] [Show and tell]
Over the last couple months, I've developed an interface for this upscaler that makes it easy to upscale entire videos. It's finally in a good spot where I feel it's appropriate to share.

It uses the portable setup included in this repo, along with ffmpeg. All you need is Python installed on your system to run it, all other files are downloaded automatically.

Please check it out! https://github.com/Nenotriple/R-ESRGAN-AnimeVideo-UI

**Comment by unamed666** [comment]
This is so cool, can u tell me how to make UI like this and what app I need?

**Comment by Nenotriple** [comment]
Thank you @eroge69

If you're interested in creating your own GUI apps, you can get started easily with Python.

Python comes with Tkinter, which is what I used to create the UI. Tkinter is an older tool (released in 1991), but it has pretty much all the features you need for a simple app. I find it easy to use, set up, and debug, and the community has created a lot of add-on modules and features that help a lot.

I also used a tool called [pygubu-designer](https://github.com/alejandroautalan/pygubu-designer) to help set up the interface. This helped me a lot when I wasn't familiar with what each widget was capable of or how different window managers work. This isn't required, but it's a handy tool, even if you just use it like a sort of sketch pad.

I use Visual Studio Code as my editor. Python is super quick to debug, and vscode has tons of helpers\extensions to make sure you're not making dumb errors, and way more.

Here's a simple tutorial that you can follow that will guide you in setting up a Tkinter window with several widgets. https://www.geeksforgeeks.org/create-first-gui-application-using-python-tkinter/

**Comment by FlM122** [comment]
which model is the best for anime videos??? to ensure that the quality is as good as possible and that the speed is high . I found AnimeJaNai, but it makes the contour lines too sharp.
